flang/lib/Lower/OpenMP.cpp

```diff
@@ -87,19 +88,35 @@ genOMP(Fortran::lower::AbstractConverter &absConv,
     auto &firOpBuilder = absConv.getFirOpBuilder();
     auto currentLocation = absConv.getCurrentLocation();
     auto insertPt = firOpBuilder.saveInsertionPoint();
+
+    // Clauses.
+    mlir::Value numThreads;
+
+    const auto &parallelOpClauseList =
+        std::get<Fortran::parser::OmpClauseList>(blockDirective.t);
+    for (const auto &clause : parallelOpClauseList.v) {
+      if (const auto &numThreadsClause =
+              std::get_if<Fortran::parser::OmpClause::NumThreads>(&clause.u)) {
+        numThreads = firOpBuilder.createIntegerConstant(
+            currentLocation, firOpBuilder.getIntegerType(32),
+            Fortran::evaluate::ToInt64(
+                *Fortran::semantics::GetExpr(numThreadsClause->v))
+                .value());
+      }
+    }
     llvm::ArrayRef<mlir::Type> argTy;
-    mlir::ValueRange range;
-    llvm::SmallVector<int32_t, 6> operandSegmentSizes(6 /*Size=*/,
-                                                      0 /*Value=*/);
-    // create and insert the operation.
+    // TODO: Add support for remaining clauses.
```

Comment:
The preference is to have the compiler abort when it reaches a TODO point in lowering.